reconcile/utils/mr/base.py

```diff
@@ -110,32 +122,63 @@ def submit_to_gitlab(self, gitlab_cli):
 
         :param gitlab_cli: The SQS Client instance.
         :type gitlab_cli: GitLabApi
-        """
-        # Avoiding duplicate MRs
-        if gitlab_cli.mr_exists(title=self.title):
-            LOG.info('MR with the same name already exists. '
-                     'Aborting MR creation.')
-            return
 
-        gitlab_cli.create_branch(new_branch=self.branch,
-                                 source_branch=self.main_branch)
+        :raises:
+            MergeRequestProcessingError: Raised when it was not possible
+              to open a MR
+        """
 
         try:
+            # Avoiding duplicate MRs
+            if gitlab_cli.mr_exists(title=self.title):
+                self.cancel(f"MR with the same name '{self.title}' "
+                            f"already exists. Aborting MR creation.")
+
+            self.ensure_tmp_branch_exists(gitlab_cli)
+
             self.process(gitlab_cli=gitlab_cli)
-        except (CancelMergeRequest, GitlabError) as details:
-            gitlab_cli.delete_branch(branch=self.branch)
-            LOG.info(details)
-            return
-
-        # Avoiding empty MRs
-        if not gitlab_cli.project.repository_compare(from_=self.main_branch,
-                                                     to=self.branch)['diffs']:
-            gitlab_cli.delete_branch(branch=self.branch)
-            LOG.info('No changes when compared to %s. Aborting MR creation.',
-                     self.main_branch)
-            return
-
-        return gitlab_cli.project.mergerequests.create(self.gitlab_data)
+
+            # Avoiding empty MRs
+            if not self.diffs(gitlab_cli):
+                self.cancel(f"No changes when compared to {self.main_branch}. "
+                            "Aborting MR creation.")
+
+            return gitlab_cli.project.mergerequests.create(self.gitlab_data)
+        except CancelMergeRequest as mr_cancel:
+            self.delete_tmp_branch(gitlab_cli)
+            LOG.info(mr_cancel)
+            # cancellation is a valid behaviour. it indicates, that the
+            # operation is not required, therefore we will not signal
+            # a problem back to the caller
+        except Exception as err:
+            self.delete_tmp_branch(gitlab_cli)
+            raise MergeRequestProcessingError(
+                f"error processing {self.name} changes "
+                f"{json.dumps(self.sqs_msg_data)} "
```

Comment:
The ldap_user and github_user integrations transport usernames via this code, etc.

I can put the payload into another attribute of the exception but the only consumer of this exception needs to log this anyways because this is vital information in the course of a bug hunt.

How would we address the need for those kinds of debug hints on one hand and the need to keep potential confidential things from making it into a log system like cloudwatch?

Reply:
I don't mind usernames too much, TBH. I worry about a bad config that throws some secret and then these logs become one extra place to expunge or review. Leave as it is and we'll review once we see some instances of this error in the logs.

Reply:
i will leave a remark in the code